### DISCUSSION TITLE
Can Federation be used similiar to Remote Schemas / Schema Stitching?

### DISCUSSION THREAD

**Discussion by debnath** [99designs/gqlgen#1792] [Q&A]
Hi, I have a little bit of a nooby question here - I am trying to get gqlgen to talk to an external GraphQL API.   Mostly the same as what issue #827 was trying to accomplish.  I don't want to copy a bunch of schemas manually, generate the code and re-implement the resolvers each time.  Ideally it could just point to the external API and pass through graphql requests.  I can see that [remote schemas ](https://www.graphql-tools.com/docs/remote-schemas/) offer this, but it is not supported by gqlgen. 

I was wondering if [Federation](https://gqlgen.com/recipes/federation/) is able to do this? I am guessing not, but would like to confirm. I went through the documentation and it looks like you can specify a service list, which initially looked like a possible elegant solution. 

```
const gateway = new ApolloGateway({
    serviceList: [
        { name: 'accounts', url: 'http://localhost:4001/query' },
        { name: 'products', url: 'http://localhost:4002/query' },
        { name: 'reviews', url: 'http://localhost:4003/query' }
    ],
});
```
But the documentation mentions each service has to be a gqlgen package... so I am thinking it is not possible!

**Comment by v9ai** [comment]
Hello, @debnath it's possible, take a look at this repo:

https://github.com/nicolad/go-next-apollo-federation
